File: Delete and Earn.java
class Solution {
    public int deleteAndEarn(int[] nums) {
        int inc=0;
        int exc=0;
        int count[]=new int[1001];
        for(int val: nums){
            count[val]++;
        }
        for(int i=0;i<=1000;i++){
            int ni=exc+count[i]*i;
            int ne=Math.max(inc,exc);
            
            inc=ni;
            exc=ne;
        }
        return Math.max(inc,exc);
    }
}

File: Delete and Earn.cpp
class Solution {
public:
    int deleteAndEarn(vector<int>& nums) {
        vector<int>dp(10001,0);
        for(auto x: nums){
            dp[x]+=x;
        }
        int val=*max_element(nums.begin(),nums.end());
        for(int i=2;i<=val;i++){
            dp[i]=max(dp[i-2]+dp[i],dp[i-1]);
        }
        return dp[val];
    }
};
